subprojects/core/src/main/java/org/gradle/api/internal/changedetection/rules/AbstractNamedFileSnapshotTaskStateChanges.java

```diff
@@ -55,6 +55,10 @@ protected AbstractNamedFileSnapshotTaskStateChanges(String taskName, TaskExecuti
         this.title = title;
         this.fileProperties = fileProperties;
         this.normalizationStrategy = normalizationStrategy;
+        retakeSnapshots();
+    }
+
+    protected void retakeSnapshots() {
         this.currentSnapshots = buildSnapshots(taskName, snapshotterRegistry, title, fileProperties);
```

Comment:
Good point about having a test. And indeed, this should take an empty snapshot. Let's handle this directly on `release`, shall we?